packages/wasm/crate/src/storage.rs

```diff
@@ -386,6 +389,63 @@ impl IndexedDBStorage {
             Err(WasmError::Anyhow(Error::msg("could not find reserves")))
         }
     }
+
+    pub async fn get_delegation_assets(&self) -> WasmResult<BTreeMap<Id, DelegationToken>> {
+        let idb_tx = self.db.transaction_on_one(&self.constants.tables.assets)?;
+        let store = idb_tx.object_store(&self.constants.tables.assets)?;
+        let all_metadata = store
+            .get_all()?
+            .await?
+            .into_iter()
+            .map(serde_wasm_bindgen::from_value)
+            .collect::<Result<Vec<Metadata>, _>>()?;
+
+        let mut assets: BTreeMap<Id, DelegationToken> = BTreeMap::new();
+
+        for metadata in all_metadata {
+            if let Ok(token) = DelegationToken::try_from(metadata.clone()) {
+                assets.insert(metadata.id(), token);
+            }
+        }
+
+        Ok(assets)
+    }
+
+    pub async fn get_notes_for_voting(
+        &self,
+        address_index: Option<keys::v1::AddressIndex>,
+        votable_at_height: u64,
+    ) -> WasmResult<Vec<(SpendableNoteRecord, IdentityKey)>> {
+        let delegation_assets = self.get_delegation_assets().await?;
+
+        let all_notes = self
+            .get_notes(NotesRequest {
+                include_spent: true,
+                address_index,
+                asset_id: None,
+                amount_to_spend: None,
+            })
+            .await?;
+
+        let mut notes_for_voting: Vec<(SpendableNoteRecord, IdentityKey)> = vec![];
+
+        for record in all_notes {
+            let Some(delegation_token) = delegation_assets.get(&record.note.asset_id()) else {
+                continue;
+            };
+
+            // Ensure if it's been spent, it is later than votable vote height
+            if let Some(height_spent) = record.height_spent {
+                if height_spent < votable_at_height {
+                    continue;
+                }
+            }
```

Comment:
Are you sure this will work for unspent notes? 
If height_spent is not set, it is serialized as 0 (maybe this is only true for typescript)

Reply:
This is relying upon the existing NotesRequest handler (https://github.com/penumbra-zone/web/blob/08ba3d65a0fc8f3a6ba2e4e5beafa2b183178ac3/packages/wasm/crate/src/storage.rs#L145-L147) which seems to indicate an unspent note is deserialized as a None.